app/models/mixins/deprecation_mixin.rb

```diff
@@ -18,6 +18,7 @@ def deprecate_belongs_to(old_belongs_to, new_belongs_to)
     def deprecate_attribute(old_attribute, new_attribute)
       deprecate_attribute_methods(old_attribute, new_attribute)
       virtual_attribute(old_attribute, -> { type_for_attribute(new_attribute.to_s) })
+      private_attribute(old_attribute)
```

Comment:
I'm not sure about this part...marking a previously public attribute as deprecated shouldn't suddenly hide it.  They should both be public.

However, if we are deprecating an already private attribute, then both old and new should be private.

Reply:
Also, can you add specs for this?

Reply:
a deprecated attribute says we are going away from an attribute. so we should stop advertising that people can use this attribute. no?
They can still use it. it just isn't advertised.

If we don't do this, how do you propose we steer people away from using deprecated attributes? The current UI (e.g.: reports, expression editor, rest docs, API query) shows these attributes as useable and a good choice.

Reply:
> They can still use it. it just isn't advertised.

Can they?  I assumed a private attribute was no longer accessabily publicly either.